src/main/java/org/folio/rest/impl/UsersAPI.java

```diff
@@ -468,30 +483,100 @@ public void postUsersOutboxProcess(Map<String, String> okapiHeaders, Handler<Asy
 
     Date now = new Date();
     entity.setCreatedDate(now);
     entity.setUpdatedDate(now);
-    PgUtil.put(TABLE_NAME_USERS, entity, entity.getId(), okapiHeaders, vertxContext, PutUsersByUserIdResponse.class, reply -> {
-      if (isDuplicateUsernameError(reply)) {
-        asyncResultHandler.handle(
-          succeededFuture(PutUsersByUserIdResponse
-            .respond400WithTextPlain(
-              "User with this username already exists")));
-        return;
-      }
-      if (isDuplicateBarcodeError(reply)) {
-        asyncResultHandler.handle(
-          succeededFuture(PutUsersByUserIdResponse
-            .respond400WithTextPlain(
-              "This barcode has already been taken")));
-        return;
-      }
-      logger.debug("Save successful");
-      asyncResultHandler.handle(reply);
-    });
+
+    pgClient.withTrans(conn -> usersService.getUserByIdForUpdate(conn, entity.getId())
+        .compose(userFromStorage -> {
+          if (userFromStorage == null) {
+            return succeededFuture(PutUsersByUserIdResponse.respond404WithTextPlain(entity.getId()));
+          }
+          boolean isConsortiaFieldsUpdated = isConsortiumUserFieldsUpdated(entity, userFromStorage);
+
+          return usersService.updateUser(conn, entity)
+            .compose(user -> {
+              if (isConsortiaFieldsUpdated) {
+                return userOutboxService.saveUserOutboxLog(conn, user, UserEvent.Action.EDIT, okapiHeaders);
+              }
+              return Future.succeededFuture();
+            })
+            .map(aVoid -> PutUsersByUserIdResponse.respond204())
+            .map(Response.class::cast);
+        })
+      .onComplete(reply -> {
+
+        if (reply.cause() != null) {
+          handleUpdateUserFailures(entity, asyncResultHandler, reply);
+          return;
+        }
+
+        userOutboxService.processOutboxEventLogs(vertxContext.owner(), okapiHeaders);
+        asyncResultHandler.handle(reply);
+      }));
+  }
+
+  private void handleUpdateUserFailures(User user, Handler<AsyncResult<Response>> asyncResultHandler, AsyncResult<Response> reply) {
+    String errorMessage = reply.cause().getMessage();
+    if (isDuplicateUsernameError(errorMessage)) {
+      logger.error("User with this username {} already exists", user.getUsername());
+      asyncResultHandler.handle(
+        succeededFuture(PutUsersByUserIdResponse
+          .respond400WithTextPlain(
+            "User with this username already exists")));
+      return;
+    }
+
+    if (isDuplicateBarcodeError(errorMessage)) {
+      logger.error("This barcode {} has already been taken", user.getBarcode());
+      asyncResultHandler.handle(
+        succeededFuture(PutUsersByUserIdResponse
+          .respond400WithTextPlain(
+            "This barcode has already been taken")));
+      return;
+    }
+
+    if (reply.cause() instanceof PgException) {
+      String errorMsg = ((PgException) reply.cause()).getDetail();
+      logger.error(errorMsg);
```

Comment:
Better to povide more context, for example like this: logger.error("DB error thrown with message: {}" errorMsg)